pkg/helm/helm.go

```diff
@@ -300,12 +304,18 @@ func (c *Client) Template(ctx context.Context, chart string, opts TemplateOption
 	// to set it ourselves.
 	client.ReleaseName = releaseName
 
+	previous := os.Getenv("HELM_CONFIG_HOME")
+	os.Setenv("HELM_CONFIG_HOME", path.Join(c.configHome, "helm-config"))
```

Comment:
Do we need to set this? I don't see anything reading this value

Reply:
I was thinking that locate chart would work with
```
	chart, err = client.ChartPathOptions.LocateChart(chart, &cli.EnvSettings{
		RegistryConfig:   envOr("HELM_REGISTRY_CONFIG", helmpath.ConfigPath("registry/config.json")),
		RepositoryConfig: envOr("HELM_REPOSITORY_CONFIG", helmpath.ConfigPath("repositories.yaml")),
		RepositoryCache:  envOr("HELM_REPOSITORY_CACHE", helmpath.CachePath("repository")),
		Debug:            true,
	})
```

apparently the `c.configHome` needs to be injected into tree of those environment variables.

Reply:
Found the problem. Chart repository is relaying on env var. Here is the stack trace:

https://github.com/helm/helm/blob/15f76cf83c670a329b62c2b5ddeb0864ec99daec/pkg/repo/chartrepo.go#L78
https://github.com/helm/helm/blob/15f76cf83c670a329b62c2b5ddeb0864ec99daec/pkg/helmpath/home.go#L24
https://github.com/helm/helm/blob/15f76cf83c670a329b62c2b5ddeb0864ec99daec/pkg/helmpath/lazypath.go#L60